### PR TITLE
Dynamodb_schema_update

### PR DESCRIPTION
**Pull request by pritamps**
Updated Schema for Dynamo DB to mimic expected use case patterns for dynamo db.

## Use cases

- Get student quiz report (SQR)
- Get all SQRs for student
- Get all SQRs for quiz 
- Get all SQRs by batch

## DynamoDB Schema

Partition key:  `session_id`
Sort key: `user_id#subject`

### Use cases

- Get student quiz report (SQR): Query by partition key and sort key
- Get all SQRs for student: `begins_with(user_id)`
- Get all SQRs for quiz: Query by partition key
- Get all SQRs for batch: Get all SQRs for quiz and then filter by batch

### CODE REVIEW

File: app/routers/reports.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            overall_performance = {}
+            for section in data:
+                parsed_section_data = _parse_section_data(section)
+                if (section["section"] == "overall"):
+                    overall_performance = parsed_section_data
+                    report_data["percentage"] = parsed_section_data["table_data"]["Percentage"]
+                    report_data["test_name"] = section["test_name"]
+                    report_data["test_date"] = section["start_date"]
+                else:
+                    section_reports.append(_parse_section_data(section))
```
Comment (suryabulusu) [inline app/routers/reports.py:68]: `section_reports.append(parsed_section_data)`

File: app/routers/student_quiz_reports.py [hunk 2]
```diff
@@ -20,12 +20,12 @@ def index_route():
         def create_student_quiz_report(report_data: CreateStudentQuizReportModel):
             return self.__student_quiz_reports_controller.create_student_quiz_report(report_data)
 
-        @api_router.get('/get')
+        @api_router.get('/get/{student_quiz_report_uid}')
```
Comment (suryabulusu) [inline app/routers/student_quiz_reports.py:23]: this should be `session_id` ?

Comment (pritamps) [reply app/routers/student_quiz_reports.py:23]: Yup -- this takes session_id and student_id as inputs now. Removed.

File: app/routers/reports.py (hunk 1 above)
Comment (pritamps) [reply app/routers/reports.py:68]: Yup!

**Pull request merged**